src/controller/python/chip/clusters/Command.py

```diff
@@ -388,7 +388,7 @@ def SendBatchCommands(future: Future, eventLoop, device, commands: List[InvokeRe
     transaction = AsyncBatchCommandsTransaction(future, eventLoop, responseTypes)
     ctypes.pythonapi.Py_IncRef(ctypes.py_object(transaction))
 
-    return builtins.chipStack.Call(
+    return await builtins.chipStack.CallAsync(
```

Comment:
I am trying to understand what is the difference here that you are trying to achieve? I see that your new `CallAsync` is functionally doing the same thing that `Call` was doing. Was this simply a way to test out that your change works as expected?

Reply:
The main difference is that `AsyncioCallableHandle` is using an `asyncio.Future` to wait on. `asyncio.Future` is non-blocking, and makes sure the asyncio event loop can continue to work on other tasks until the future is done.

Now the value of this really depends how long it takes until `AsyncioCallableHandle.__call__` is called (which is called from the Matter thread via `PlatformManager::ScheduleWork`), or in other words, how busy the Matter thread is, plus how long it actually takes to execute the function on the Matter thread.

But in any case, such OS level context switches are considered relatively slow, and asyncio would like to go on to other tasks meanwhile.

In my testing, most function calls were relatively quick. But a small function did get scheduled before the done task, as in:

```python
async def _test():
    print("Do other work")

asyncio.create_task(_test())
print("Calling SDK")
await CallAsync(...)
print("Called SDK")
```

Lead to:

```
Calling SDK
Do other work
Called SDK
```

Which tells me that the asyncio event loop got processed.

Reply:
Prior to calling `CallAsync(...)` what did you see with `Call(...)`, so with something like
```
async def _test():
    print("Do other work")

asyncio.create_task(_test())
print("Calling SDK")
Call(...)
print("Called SDK")
```

Are you trying to prevent:
```
Calling SDK
Called SDK
Do other work
```

Overall I am happy with the changes you have here as I think moving to asyncio and cleaning up this code is general is beneficial. But I am trying to understand why these calls are switching from `builtins.chipStack.Call(` to `await builtins.chipStack.CallAsync(`

My understanding of `Call()` after code audit, is that the docstring is completely wrong as it is not a blocking call. It simply schedules a code to be run on the Matter thread, but will return before the call actually is completed. Are we saying that these should have always been blocking calls until the scheduled work was completed?

Reply:
> Are you trying to prevent:
> 
> ```
> Calling SDK
> Called SDK
> Do other work
> ```

Essentially yes. We need to wait for at least two Thread context switches, which is an eternity for pure execution time.  This order really shows that the event loop is being blocked. It is only short (as per my testing 130us for a quick function), but it really undermines the idea of asyncio. 

> My understanding of Call() after code audit, is that the docstring is completely wrong as it is not a blocking call. It simply schedules a code to be run on the Matter thread, but will return before the call actually is completed. Are we saying that these should have always been blocking calls until the scheduled work was completed?

No, it calls `Wait` on the call object, which is essentially calling Wait on the per-call `threading.Condition` object. So this is truly a synchronous call, waiting for the function to complete execution on the Matter thread.

Reply:
> No, it calls `Wait` on the call object, which is essentially calling Wait on the per-call `threading.Condition` object.

Right! Thanks for that. I lost that information over the weekend.

Let me confirm my understanding: The main reason we are moving from `builtins.chipStack.Call(` to await `builtins.chipStack.CallAsync(` is because when we called something like devCtrl.SendCommand() that would be using the asyncio event loop, but as soon as we call `builtins.chipStack.Call(` we would be blocking the asyncio event loop. By moving to your version of `builtins.chipStack.CallAsync(` we are now no longer blocking the asyncio event loop.

Reply:
Yes, exactly. 👍